### PR TITLE
Improve Project Layout and Refactor Controller Package

### PR DESCRIPTION
**Pull request by peterkellyonline**
### Proposed changes
After the project restructure PR (this branch is branches from that one), this PR then pulls apart some of the complexity in the load balancer controller. Extracts handlers and builds them separately.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

**Comment by isserrano** [comment]
If go test ./... fixed, everything else LGTM

**Comment by isserrano** [comment]
CI fixed, build is fine

File: .gitignore [hunk 1]
```diff
@@ -24,9 +24,10 @@ Session.vim
 # Ingress Controller binaries
 osx-nginx-ingress
 nginx-ingress
+!nginx-ingress/
 osx-nginx-plus-ingress
 nginx-plus-ingress
-nginx-controller/nginx-controller
+cmd/nginx-ic/nginx-ic
```
[the remark this answers is not in the document]
Comment (peterkellyonline) [reply .gitignore]: it is needed as it is the binary produced from `go build`, but it should be nginx-ingress

[the remark this answers is not in the document]
Comment (pleshakov) [reply .gitignore]: it seems that the following lines cover that:
```
nginx-ingress
!nginx-ingress/
```

**Pull request merged**